File: app/src/at-vinhnguyen/javacore/OOP/Rectangle.java
package OOP;

public class Rectangle extends Shape {

    private float a, b;

    public Rectangle(float a, float b) {
        this.a = a;
        this.b = b;
    }

    @Override
    float Acreage() {
        return a * b;
    }

    @Override
    float Perimeter() {
        return (a + b) * 2;
    }

    @Override
    void printNameOfTheShape() {
        if (a == b) {
            super.printNameOfTheShape();
            return;
        }
        System.out.println("Rectangle:");
    }

}

File: app/src/at-vinhnguyen/java/asiantech/internship/summer/kotlincore/oop/Rectangle.kt
package asiantech.internship.summer.kotlincore.oop

class Rectangle(private val a: Float, private val b: Float) : Shape() {

  override fun Acreage(): Float {
    return a * b
  }

  override fun Perimeter(): Float {
    return (a + b) * 2
  }

  override fun printNameOfTheShape() {
    if (a == b) {
      super.printNameOfTheShape()
      return
    }
    println("Rectangle:")
  }
}
